rebuild/src/public_key.cc

```diff
@@ -0,0 +1,154 @@
+#include "public_key.h"
+
+void PublicKey::Init(
+    Napi::Env env, Napi::Object &exports, BlstTsAddon *module) {
+    Napi::HandleScope scope(env);
+    auto proto = {
+        StaticMethod(
+            "deserialize",
+            &PublicKey::Deserialize,
+            static_cast<napi_property_attributes>(
+                napi_static | napi_enumerable)),
+        InstanceMethod(
+            "serialize",
+            &PublicKey::Serialize,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "keyValidate",
+            &PublicKey::KeyValidate,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "isInfinity",
+            &PublicKey::IsInfinity,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "multiplyBy",
+            &PublicKey::MultiplyBy,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+    };
+
+    Napi::Function ctr = DefineClass(env, "PublicKey", proto, module);
+    module->_public_key_ctr = Napi::Persistent(ctr);
+    exports.Set(Napi::String::New(env, "PublicKey"), ctr);
+
+    Napi::Object js_exports = exports.Get("BLST_CONSTANTS").As<Napi::Object>();
+    js_exports.Set(
+        Napi::String::New(env, "PUBLIC_KEY_LENGTH_COMPRESSED"),
+        Napi::Number::New(env, BLST_TS_PUBLIC_KEY_LENGTH_COMPRESSED));
+    js_exports.Set(
+        Napi::String::New(env, "PUBLIC_KEY_LENGTH_UNCOMPRESSED"),
+        Napi::Number::New(env, BLST_TS_PUBLIC_KEY_LENGTH_UNCOMPRESSED));
+}
+
+Napi::Value PublicKey::Deserialize(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE(info, env, module)
+    Napi::Value pk_bytes_val = info[0];
```

Comment:
Can `info` be empty? It'd be safer to check all the expectations at the beginning of the function.

Reply:
its always passed to functions called from JS context.  its always got values and the index operator is overridden by the class to check for valid indexes.  its safe